### ISSUE TITLE
Enhancement(s): Marketplace/Extension UI Ideas

### ISSUE THREAD

**Issue by kingthorin** [zaproxy/zaproxy#3588]
A few ideas that have been brain stormed via email, all of which have potential.

- [ ] Revise the Marketplace dialog to allow multi-line display for the description column/field.
- [ ] Add a button to the dialog that would open the corresponding help info (where help info is in place).
- [ ] Add some functionality to support a kind of "First Run" dialog to display info to the user. ZAP has extensions that are used/accessed in many different ways, it would be nice if extensions could provide the user a tip or short message on their usage when first run.
- [ ] Add some functionality to support a kind of "On Install" dialog to display information to the user about changes to the add-on, etc.

**Comment by kingthorin** [comment]
I guess the first-use and on-install ideas kind of overlap and could be tackled all at once depending how they’re implemented.

**Comment by thc202** [comment]
Agreed, worth describe how that should work (e.g. where's shown, which info and from where).

**Comment by psiinon** [comment]
Additional suggestion - include an optional link to the source code. Maybe this can be automated for the official ZAP ones?

**Comment by kingthorin** [comment]
I was picturing an additional dialog after completion of install. Though I guess text in the log and output tab might work (though less user friendly). I was picturing the content coming from a specific key (or keys) in Messages.properties (so that it could be translated), but I guess it could also be a build property (like status/version), but that would limit us to en-GB I guess.
